### PR TITLE
mat64: add reuseAsZeroed and use in Exp

### PR DESCRIPTION
**Pull request by kortschak**
~~WIP: Waiting to add use in QFromQR.~~

### CODE REVIEW

**Comment by btracey** [comment]
I guess this is okay, but I would be happier if there was less copying from reuseAs. I don't see a good way around it though.

LGTM

**Comment by kortschak** [comment]
This PR reduces copying when complete (after the QFromQR use is in). There is the option to merge reuseAs and reuseAsZeroed by passing a bool, but I don't like the readability impact of that (which is why both use and useZeroed exist).

**Comment by btracey** [comment]
Yea, agreed on both accounts. Maybe add a comment to reuseAs that if that implementation changes, reuseAsZeroed should change as well?

**Comment by kortschak** [comment]
Will do.

**Comment by kortschak** [comment]
OK, this is ready.

PTAL

File: mat64/dense.go [hunk 1]
```diff
@@ -83,6 +85,35 @@ func (m *Dense) reuseAs(r, c int) {
...
+		}
+		m.capRows = r
+		m.capCols = c
+		return
+	}
+	if r != m.mat.Rows || c != m.mat.Cols {
+		panic(matrix.ErrShape)
+	}
+	for i := 0; i < r; i++ {
+		zero(m.mat.Data[i*m.mat.Stride : i*m.mat.Stride+r])
```
Comment (btracey) [inline mat64/dense.go]: i*m.mat.Stride + c, right?

**Comment by btracey** [comment]
LGTM with fix.

File: mat64/dense.go (hunk 1 above)
Comment (kortschak) [inline mat64/dense.go]: Indeed.

Comment (kortschak) [inline mat64/dense.go]: Interestingly it doesn't show in tests because both uses are for creating I, so r == c.

**Pull request merged**